src/accordion/accordion.ts

```diff
@@ -59,53 +62,130 @@ export class NgbPanel {
    */
   @Input() title: string;
 
+  @ContentChild(NgbPanelContent) contentTpl: NgbPanelContent;
+  @ContentChild(NgbPanelTitle) titleTpl: NgbPanelTitle;
+
   constructor(@Optional() @Inject(forwardRef(() => NgbAccordion)) private accordion: NgbAccordion) {}
 
-  toggleOpen(event): void {
-    event.preventDefault();
-    if (!this.disabled) {
-      this.open = !this.open;
-      if (this.open && this.accordion) {
-        this.accordion.closeOthers(this);
-      }
+  ngOnChanges(changes: SimpleChanges) {
+    // when 'open' changes to 'true' we potentially have to close other panels
+    let openChange = changes['open'];
+    if (this.accordion && openChange && !openChange.isFirstChange() && openChange.currentValue) {
+      this.accordion.closeOthers(this.id);
     }
   }
 }
 
 /**
+ * The payload of the panel change event
+ */
+export interface NgbPanelChangeEvent {
+  panelId: string;
+  nextState: boolean;
+  preventDefault();
+}
+
+/**
  * The NgbAccordion directive is a collection of panels.
  * NgbAccordion can assure that only panel can be opened at a time.
  */
 @Component({
   selector: 'ngb-accordion',
-  host: {'role': 'tablist', '[attr.aria-multiselectable]': '!closeOtherPanels'},
-  template: `<ng-content></ng-content>`
+  exportAs: 'ngbAccordion',
+  template: `
+    <div *ngFor="let panel of _panels">
+      <div class="panel panel-default" [class.panel-open]="panel.open">
+        <div class="panel-heading" role="tab" [id]="panel.id">
+          <h4 class="panel-title">    
+            <a tabindex="0" (click)="toggle(panel.id)" [class.text-muted]="panel.disabled">
+              {{panel.title}}<template [ngTemplateOutlet]="panel.titleTpl?.templateRef"></template>          
+            </a>
+          </h4>
+        </div>
+        <div class="panel-body" *ngIf="panel.open">
+          <template [ngTemplateOutlet]="panel.contentTpl.templateRef"></template>
+        </div>
+      </div>
+    </div>
+`
 })
-export class NgbAccordion implements AfterContentChecked {
+export class NgbAccordion implements OnChanges,
+    AfterContentInit {
   @ContentChildren(NgbPanel) _panels: QueryList<NgbPanel>;
 
   /**
+   * An array of panel of identifiers that should be opened
+   */
+  @Input() activeIds: string[] = [];
```

Comment:
Could we have it as `string | string[]` so for `closeOthers` case people don't have to specify an array?